core/src/epicli/data/common/ansible/playbooks/roles/vault/files/vault/configure-vault.sh

```diff
@@ -0,0 +1,294 @@
+      touch $token_path;
+      chmod 0640 $token_path;
+    fi
+    local users_path_response=$(curl -o -I -L -s -w "%{http_code}" --header "X-Vault-Token: $token" --request LIST "$VAULT_ADDR/v1/auth/userpass/users");
+    curl --header "X-Vault-Token: $token" --request LIST "$vault_addr/v1/auth/userpass/users" | jq -e ".data.keys[] | select(.== \"$username\")";
+    local local command_result="$?";
+    echo "Command result: $command_result";
+    echo "override_existing_vault_users: $override_existing_vault_users";
+    if [ "${override_existing_vault_users,,}" = "true" ] || [ $users_path_response -eq 404 ] || [ "$command_result" = "4" ]; then
+        log_and_print "Creating user: $username...";
+        local password="$( < /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c32 )";
+        vault write auth/userpass/users/$username password=$password policies=$policy;
+        check_vault_error "$?" "User: $username created." "There was an error during creation of user: $username.";
+        echo "$username;$policy;$password;" >> $token_path;
+    elif [ "$command_result" = "0" ]; then
+        log_and_print "$username already exists. Not adding or modyfing.";
+        echo "$username;$policy;ALREADY_EXISTS;" >> $token_path;
+    else
+        exit_with_error "There was an critical error during adding user $username.";
+    fi
+}
+
+function create_vault_users_from_file {
+    local vault_install_path="$1";
+    local users_file_csv_path="$vault_install_path/users.csv";
+    local users_token_path="$vault_install_path/tokens-`date +"%Y-%m-%d-%H%M%S"`.csv";
+    local token="$2";
+    local vault_addr="$3";
+    local override_existing_vault_users="$4";
+    cat $users_file_csv_path | while read -r line ; do
+        local username="$( echo $line | cut -d ';' -f 1 )";
+        local policy="$( echo $line | cut -d ';' -f 2 )";
+        create_vault_user "$username" "$policy" "$users_token_path" "$token" "$vault_addr" "$override_existing_vault_users";
+    done
+}
+
+function cleanup {
+    rm -f "$HOME/.vault-token";
+}
+
+while getopts ":a:c:h?" opt; do
+    case "$opt" in
+        a) VAULT_IP=$OPTARG;;
+        c) CONFIG_FILE=$OPTARG;;
+        ? | h | *) print_help; exit 2;;
+    esac
+done
+
+if [ $OPTIND -eq 1 ]; then
+    print_help;
+    exit_with_error "No options passed to script. Aborting.";
+fi
+
+source "$CONFIG_FILE";
+
+INIT_FILE_PATH="$VAULT_INSTALL_PATH/init.txt"
+VAULT_CONFIG_DATA_PATH="$VAULT_INSTALL_PATH/config"
+export VAULT_ADDR="http://$VAULT_IP:8200"
+export KUBECONFIG=/etc/kubernetes/admin.conf
+PATH=$VAULT_INSTALL_PATH/bin:/usr/local/bin/:$PATH
+
+if [ "${VAULT_TOKEN_CLEANUP,,}" = "true" ] ; then
+    trap cleanup EXIT INT TERM;
+fi
+
+initialize_vault "$INIT_FILE_PATH";
+
+if [ "${SCRIPT_AUTO_UNSEAL,,}" = "true" ] ; then
+    unseal_vault "$INIT_FILE_PATH";
+fi
+
+check_if_vault_is_unsealed;
+
+log_and_print "Logging into Vault.";
+LOGIN_TOKEN="$(grep "Initial Root Token:" "$INIT_FILE_PATH" | awk -F'[ ]' '{print $4}')";
```

Comment:
What if this command fails and token is undefined? Wouldn't this be better to check if before running the vault binary?